### PR TITLE
Onlyauth

### PR DESCRIPTION
**Pull request by thinklikeadesigner**
This commit only has the authorization with GitHub integration. No queries.

## New API endpoints
```api/auth/login``` for auth0 social login
```api/auth/logout```  for logging user out
```api/auth/github```  for access token _(prereq is logging in)_

### CODE REVIEW

File: pages/api/hello.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import { fromUrl } from "hosted-git-info";
-import fetch from "node-fetch";
+import { fromUrl } from 'hosted-git-info';
+import fetch from 'node-fetch';
```
Comment (DianaSuvorova) [inline pages/api/hello.js:2]: Hey @thinklikeadesigner, I'd rather not do this as part of this diff. Not sure if this is intentional change also. DO you prefer single quotes over double?

In any case would make it a different change if feel strongly about it

File: pages/login.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+export default function Login() {
```
Comment (DianaSuvorova) [inline pages/login.js:1]: I would say we don't even need a page. Just `/login, /logout` etc working. I don't think we need a separate page with a button.

File: pages/logout.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+export default function Logout() {
```
Comment (DianaSuvorova) [inline pages/logout.js:1]: same, let's just create auth routes. No need  to add them as separate UI elements

Comment (thinklikeadesigner) [reply pages/logout.js]: done

File: pages/login.js (hunk 2 above)
Comment (thinklikeadesigner) [reply pages/login.js]: did you want me to delete the profile page too?

File: pages/api/hello.js (hunk 1 above)
Comment (thinklikeadesigner) [reply pages/api/hello.js:2]: air bnb style guide is single quotes, so that is what my eslint is set at

Comment (DianaSuvorova) [reply pages/api/hello.js:2]: That's cool to set it up this way. Please make this change as a separate PR. Don't do it as part auth diff.
Also please make sure it is part of this project's eslint config. It shouldn't be part of local/dev env preference. I will be happy to help you with that. It should be separate change.

File: pages/about/index.js [hunk 4]
```diff
@@ -13,34 +13,30 @@ export default function About() {
         <div className={styles.about__content}>
           <p>Hi There!</p>
           <p>
-            We are a group of people who believe that technology and its
-            community fuel the development and growth of each other.
+            We are a group of people who believe that technology and its community fuel the
```
Comment (DianaSuvorova) [inline pages/about/index.js:16]: this looks like an old version of this text. maybe some issue with rebase?

File: pages/_app.js [hunk 5]
```diff
@@ -1,9 +1,12 @@
...
+import React from 'react';
+import { UserProvider } from '@auth0/nextjs-auth0';
 
-// eslint-disable-next-line react/prop-types
-function MyApp({ Component, pageProps }) {
-  // eslint-disable-next-line react/jsx-props-no-spreading
-  return <Component {...pageProps} />;
+export default function App({ Component, pageProps }) {
+  return (
+    <UserProvider>
```
Comment (DianaSuvorova) [inline pages/_app.js:8]: This is unnecessary since we don't surface user auth on the UI yet. We will be using for querying only

**Review by DianaSuvorova — COMMENTED** [review]
All style related change should be its own diff.
And yes, I think we don't need any new pages/buttons for auth/login yet. Just so URL works correctly.

File: package-lock.json [hunk 6]
```diff
@@ -1,9 +1,7297 @@
 {
   "name": "ufo-explorer",
   "version": "0.1.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
Comment (DianaSuvorova) [inline package-lock.json:4]: what's your node version? This change should be ok.. Although we should pin node version so it does not switch between dev env

**Review by DianaSuvorova — APPROVED** [review]

**Pull request merged**